src/Microsoft.Health.Dicom.Core.UnitTests/Features/Query/QueryParserTests.cs

```diff
@@ -436,6 +436,14 @@ public void GivenDateTag_WithInvalidDate_Throw(string key, string value)
                 .Parse(CreateParameters(GetSingleton(key, value), QueryResource.AllSeries), QueryTagService.CoreQueryTags));
         }
 
+        [Theory]
+        [InlineData("ReferencedRequestSequence.Requested​Procedure​ID", "Foo")]
+        public void GivenSequenceTag_WithMultipleTagsNotSupported_ThenThrow(string key, string value)
```

Comment:
Do you have tests to check parameters allowed max, say on limit param?

Reply:
Yes, E2e tests are added to check for bad request